server/block/cactus.go

```diff
@@ -25,31 +25,17 @@ func (c Cactus) UseOnBlock(pos cube.Pos, face cube.Face, _ mgl64.Vec3, w *world.
 	if !used {
 		return false
 	}
-	_, ok := w.Block(pos.Side(cube.FaceDown)).(Cactus)
-	if !supportsVegetation(c, w.Block(pos.Side(cube.FaceDown))) && !ok {
+	if !c.CanGrowHere(pos, w) {
 		return false
 	}
-	for _, face := range cube.HorizontalFaces() {
-		if _, ok := w.Block(pos.Side(face)).(Air); !ok {
-			return false
-		}
-	}
 
 	place(w, pos, c, user, ctx)
 	return placed(ctx)
 }
 
 // NeighbourUpdateTick ...
 func (c Cactus) NeighbourUpdateTick(pos, _ cube.Pos, w *world.World) {
-	for _, face := range cube.HorizontalFaces() {
-		if _, ok := w.Block(pos.Side(face)).(Air); !ok {
-			w.SetBlock(pos, nil, nil)
-			w.AddParticle(pos.Vec3Centre(), particle.BlockBreak{Block: c})
-			return
-		}
-	}
-	_, ok := w.Block(pos.Side(cube.FaceDown)).(Cactus)
-	if !supportsVegetation(c, w.Block(pos.Side(cube.FaceDown))) && !ok {
+	if !c.CanGrowHere(pos, w) {
 		w.SetBlock(pos, nil, nil)
 		w.AddParticle(pos.Vec3Centre(), particle.BlockBreak{Block: c})
 	}
```

Comment:
Looks like it was missing before, but could you make it drop the cactus here as well?

Reply:
added dropItem().